src/reaction_network.jl

```diff
@@ -1,45 +1,4 @@
 ### Temporary deprecation warning - Eventually to be removed. ###
-deprication_message = """
-@reaction_network notation where parameters are declared after "end", e.g. like:
-
-```julia
-@reaction_network begin
-    p, 0 --> X
-    d, X --> 0
-end p d
-```
-
-has been deprecated in favor of a notation where the parameters are inferred, e.g:
-
-```julia
-@reaction_network begin
-    p, 0 --> X
-    d, X --> 0
-end
-```
-
-Parameters and species can be explicitly indicated using the @parameters and @species
-macros, e.g:
-
-```julia
-@reaction_network begin
-    @parameters p d
-    @species X(t)
-    p, 0 --> X
-    d, X --> 0
-end
-```
-"""
-macro reaction_network(name::Symbol, ex::Expr, parameters...)
-    error(deprication_message)
-end
-macro reaction_network(name::Expr, ex::Expr, parameters...)
-    error(deprication_message)
-end
-macro reaction_network(ex::Expr, parameters...)
-    error(deprication_message)
-end
-
```

Comment:
So, we could keep these around. The reason I had them removed is that now when we have both `@reaction_network` and `@network_component`, it seems best that the former calls the latter (but with a `complete` call), and the latter does the actual stuff. When I tried to implement, I had a problem that some of these were more specific, and where called instead. Hence I removed them.

They have been around since the change, so maybe it makes sense to remove them anyway (although might be useful if someone learns from an old tutorial).

When we figured out the new macro implementation, I don''t mind adding these back in, if possible.

Reply:
This issue seems realted to this one: https://github.com/JuliaLang/julia/issues/37691